.github/workflows/tag_and_publish.yaml

```diff
@@ -0,0 +1,43 @@
+name: Tag and Publish Release
+on:
+  push:
+    branches:
+      - main
+
+concurrency: ${{ github.ref }}
+
+jobs:
+  tag-release:
+    runs-on: ubuntu-18.04
```

Comment:
@TheRealFalcon Two other cases where we probably want to bump off of ubuntu-18.04 -> ubuntu-latest in our  workflows.